track_info.c

```diff
@@ -201,7 +201,7 @@ int track_info_matches_full(const struct track_info *ti, const char *text,
 				matched = 1;
 		}
 
-		if (flags_set(ti, exclude_flags) && match_word(ti, word, exclude_flags))
+		if (match_word(ti, word, exclude_flags))
```

Comment:
Why don't we call `flags_set` here?

Reply:
It's just not needed because `match_word()` does the same checks anyway. This line isn't fixing anything, I just thought it was clearer to understand.

Reply:
I think the same goes for call to `flags_set` above: they should either both be present, or both be absent for consistency. Probably `flags_set` was used as an optimization, to avoid any string comparisons, but I am not sure.

Reply:
The first one is needed though, to distinguish "no applicable tags -- fallback to filename search" from "no match -- don't fall back to filename".